chapter_2/CMakeLists.txt

```diff
@@ -3,31 +3,26 @@
 # SPDX-License-Identifier: MIT
 
 project(
-  "CMake Tips and Tricks Chapter 2"
-  VERSION 1.0
-  DESCRIPTION "A simple C++ project to demonstrate basic CMake usage"
-  LANGUAGES CXX)
-
-# Set C++ standard to 11 for all subdirectories (component1, component2,
-# driver_app) in this folder.
-set(CMAKE_CXX_STANDARD 11)
-# Make C++ 11 standard a must. Setting CMAKE_CXX_STANDARD is a *should* by
-# default. When this variable is set and selected compiler does not support
-# C++11, the CMake configure step will fail.
-set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+    "CMake Tips and Tricks Chapter 2"
+    VERSION 1.0
+    DESCRIPTION "A simple C++ project to demonstrate basic CMake usage"
+    LANGUAGES CXX
+)
 
 # Option to exclude driver application from build.
 set(CHAPTER2_BUILD_DRIVER_APPLICATION
     TRUE
-    CACHE BOOL "Whether to include driver application in build. Default: True")
+    CACHE BOOL "Whether to include driver application in build. Default: True"
+)
 # Hide this option from GUI's by default.
 mark_as_advanced(CHAPTER2_BUILD_DRIVER_APPLICATION)
 
 # MSVC requires these two flags to be set in order to link driver executable
-# against shared library
+# against shared library.
+# The topic is about symbol visibility will be further discussed in Chapter 3.
 if(MSVC)
-  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
-  set(BUILD_SHARED_LIBS TRUE)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+    set(BUILD_SHARED_LIBS TRUE)
```

Comment:
If symbol visibility and the general messiness of cross-platform shared libraries using C++ are already covered as part of chapter 3, then I would suggest just removing this and using static libraries for the components. The hardcoding of `BUILD_SHARED_LIBS` also pointless, because the components all explicitly declare their library type, so the variable is unused. `MSVC` does not mean correctly mean Windows target platform either. I personally use LLVM Clang on Windows and for that `MSVC` is not set, but I still could use `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` from the command line to turn the shared libraries inside-out and expose its guts. The correct way to query the target platform is the [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html) variable, which will be set to `Windows` for a toolchain that is targetting Windows.